src/App.js

```diff
@@ -1,15 +1,74 @@
 import React from 'react';
-import { Switch, BrowserRouter as Router, Route } from 'react-router-dom';
+import ForConsumers from './pages/ForConsumers';
+import ForContractors from './pages/ForContractors';
 import LandingPage from './pages/LandingPage';
+import LoggedInConsumer from './pages/LoggedInConsumer';
+import LoggedInContractor from './pages/LoggedInContractor';
+import { useAuth0 } from '@auth0/auth0-react';
+import { Switch, BrowserRouter as Router, Route } from 'react-router-dom';
 
 const App = () => {
-    return (
-        <Router>
-            <Switch>
-                <Route exact path="/" component={LandingPage}></Route>
-            </Switch>
-        </Router>
-    );
+    const { isAuthenticated, isLoading, user } = useAuth0();
+
+    //TODO: REFACTOR THESE IF STATEMENTS
+    if (isLoading) {
+        return <div>loading 123...</div>;
+    }
+    if (!isAuthenticated) {
+        return (
+            <Router>
+                <Switch>
+                    <Route exact path="/" component={LandingPage}></Route>
+                    <Route
+                        exact
+                        path="/ForConsumers"
+                        component={ForConsumers}
+                    ></Route>
+                    <Route
+                        exact
+                        path="/ForContractors"
+                        component={ForContractors}
+                    ></Route>
+                </Switch>
+            </Router>
+        );
+    }
+    if (
+        isAuthenticated &&
+        user &&
+        user['https://bidsdirect.com/roles'] === 'Consumer'
```

Comment:
in the auth0 rules that i have created behind the scenes, this roles link lives on the context.idtoken. there is a little bit of magic happening here because they have a outline for the rule, and i just had to massage it to get it working correctly. For a reason i dont understand this is only working if its an HTTPS link. does not matter the url, but just making it user.roles for example will not work.

Reply:
lame sauce